tests/timeseries/_formula_engine/test_formula_composition.py

```diff
@@ -171,16 +171,70 @@ async def test_formula_composition_missing_bat(self, mocker: MockerFixture) -> N
         assert count == 10
 
     async def test_formula_composition_min_max(self, mocker: MockerFixture) -> None:
-        """Test the composition of formulas with min/max values."""
+        """Test the composition of formulas with the min and max."""
         mockgrid = MockMicrogrid(grid_meter=True)
+        mockgrid.add_chps(1)
         await mockgrid.start(mocker)
 
         logical_meter = microgrid.logical_meter()
-        engine_min = logical_meter.grid_power._min(  # pylint: disable=protected-access
+        engine_min = logical_meter.grid_power.min(logical_meter.chp_power).build(
+            "grid_power_min"
+        )
+        engine_min_rx = engine_min.new_receiver()
+        engine_max = logical_meter.grid_power.max(logical_meter.chp_power).build(
+            "grid_power_max"
+        )
+        engine_max_rx = engine_max.new_receiver()
+
+        await mockgrid.mock_resampler.send_meter_power([100.0, 200.0])
+
+        # Test min
+        min_pow = await engine_min_rx.receive()
+        assert (
+            min_pow and min_pow.value and min_pow.value.isclose(Power.from_watts(100.0))
+        )
+
+        # Test max
+        max_pow = await engine_max_rx.receive()
+        assert (
+            max_pow and max_pow.value and max_pow.value.isclose(Power.from_watts(200.0))
+        )
+
+        await mockgrid.mock_resampler.send_meter_power([-100.0, -200.0])
+
+        # Test min
+        min_pow = await engine_min_rx.receive()
+        assert (
+            min_pow
+            and min_pow.value
+            and min_pow.value.isclose(Power.from_watts(-200.0))
+        )
+
+        # Test max
+        max_pow = await engine_max_rx.receive()
+        assert (
+            max_pow
+            and max_pow.value
+            and max_pow.value.isclose(Power.from_watts(-100.0))
+        )
+
+        await engine_min._stop()  # pylint: disable=protected-access
+        await mockgrid.cleanup()
+        await logical_meter.stop()
+
+    async def test_formula_composition_min_max_const(
+        self, mocker: MockerFixture
+    ) -> None:
+        """Test the compositing formulas and constants with the min and max functions."""
+        mockgrid = MockMicrogrid(grid_meter=True)
+        await mockgrid.start(mocker)
+
+        logical_meter = microgrid.logical_meter()
+        engine_min = logical_meter.grid_power.min(  # pylint: disable=protected-access
             Power.zero()
         ).build("grid_power_min")
         engine_min_rx = engine_min.new_receiver()
-        engine_max = logical_meter.grid_power._max(  # pylint: disable=protected-access
+        engine_max = logical_meter.grid_power.max(  # pylint: disable=protected-access
```

Comment:
Then you might want to remove this one too (if it works)
```suggestion
        engine_max = logical_meter.grid_power.max(
```